src/t_hash.c

```diff
@@ -1028,10 +1056,401 @@ static void addHashIteratorCursorToReply(writePreparedClient *wpc, hashTypeItera
+ *
+ * - Apply changes:
+ *   Either deletes expired fields or sets fields with optional expiration.
+ *
+ * - Clean up and notify:
+ *   Deletes the key if the hash becomes empty.
+ *   Emits keyspace notifications for changes (see below).
+ *   Modifies the command vector for AOF propagation if necessary.
+ *
+ *
+ * Return Value:
+ * - Returns integer 1 if all fields were successfully updated or deleted.
+ * - Returns integer 0 if no fields were updated due to condition failures.
+ *
+ *
+ * Keyspace Notifications (if enabled):
+ * - "hset"      — Emitted when fields are added or updated.
+ * - "hexpire"   — Emitted when expiration is set on fields.
+ * - "hexpired"  — Emitted when fields are immediately expired and deleted.
+ * - "del"       — Emitted if the entire key is removed (empty hash).
+ *
+ *
+ * Client Reply:
+ * - Integer reply: 1 if all changes succeeded, 0 if no changes occurred. */
+void hsetexCommand(client *c) {
+    robj *o;
+    robj *expire = NULL;
+    robj *comparison = NULL;
+    int unit = UNIT_SECONDS;
+    int flags = ARGS_NO_FLAGS;
+    int fields_index = 0;
+    long long num_fields = 0;
+    long long when = EXPIRY_NONE;
+    int i = 0;
+    int set_flags = HASH_SET_COPY, set_expired = 0;
+    int changes = 0;
+    robj **new_argv = NULL;
+    int new_argc = 0;
+
+    for (; fields_index < c->argc; fields_index++) {
+        if (!strcasecmp(c->argv[fields_index]->ptr, "fields")) {
+            /* checking optional flags */
+            if (parseExtendedCommandArgumentsOrReply(c, &flags, &unit, &expire, &comparison, COMMAND_HSET, fields_index++) != C_OK) return;
+            if (getLongLongFromObjectOrReply(c, c->argv[fields_index++], &num_fields, NULL) != C_OK) return;
+            break;
+        }
+    }
+    /* Check that the parsed fields number matches the real provided number of fields */
+    if (!num_fields || num_fields != (c->argc - fields_index) / 2) {
+        addReplyErrorObject(c, shared.syntaxerr);
+        return;
+    }
+
+    o = lookupKeyWrite(c->db, c->argv[1]);
+    if (checkType(c, o, OBJ_HASH))
+        return;
+
+    /* Check for object existence condition */
+    if ((flags & ARGS_SET_NX && o) || (flags & ARGS_SET_XX && !o)) {
+        addReply(c, shared.czero);
+        return;
+    }
+
+    if (o == NULL) {
+        o = createHashObject();
+        dbAdd(c->db, c->argv[1], &o);
+    }
+
+    /* Handle parsing and calculating the expiration time. */
+    if (flags & ARGS_KEEPTTL)
+        set_flags |= HASH_SET_KEEP_EXPIRY;
+    else if (expire) {
+        long long basetime = (flags & (ARGS_EXAT | ARGS_PXAT)) ? 0 : commandTimeSnapshot();
+
+        if (convertExpireArgumentToUnixTime(c, expire, basetime, unit, &when) == C_ERR)
+            return;
+
+        if ((when == basetime) || (((flags & ARGS_PXAT) || (flags & ARGS_EXAT)) && ((when == basetime) || checkAlreadyExpired(when)))) {
+            set_expired = 1;
+        }
```

Comment:
I don't quite understand this condition. Can't this condition be simplified to:
```suggestion
        set_expired = checkAlreadyExpired(when);
```

Reply:
I don't know what came over me. Yes you are 100% correct. will change all cases